### ISSUE TITLE
When do you plan on supporting rails 7

### ISSUE THREAD

**Issue by maszuDEV** [graphql-devise/graphql_devise#207]
### Question

Hello everyone! Do you know when rails 7 support will be added?

Best,

**Comment by mcelicalderon** [comment]
Hey, @maszuDEV! We still depend on [this PR](https://github.com/lynndylanhurley/devise_token_auth/pull/1517) getting into a DTA release, before we can release a new version that supports Rails 7.0. Master already supports Rails 7.0 and everything seems to work fine. We are just getting an error on the generator specs and we need to look into that. So if you want to give it a try, you can use this gem's and DTA's master branch on your gemfile. Hopefully we can release a new version soon.

**Comment by maszuDEV** [comment]
@mcelicalderon thanks for the answer! Then it's best to wait to upgrade as well.

**Comment by mcelicalderon** [comment]
Test matrix already includes Rails 7.0 and everything looks fine. We still need to wait for DTA to release a new version https://github.com/lynndylanhurley/devise_token_auth/pull/1517#issuecomment-1003751213

**Comment by openscript** [comment]
The gem seems to support Rails 7 now, but it's missing in the current release?

![image](https://user-images.githubusercontent.com/1105080/173181973-108e43e6-b6ee-4dae-acd2-f9958f0f8841.png)

**Comment by mcelicalderon** [comment]
Hey @openscript! As stated in https://github.com/graphql-devise/graphql_devise/issues/207#issuecomment-1021243045 DTA is one of the main dependencies of this gem and they still haven't released a version that supports rails 7. Support for rails 7 is already merged into master for both DTA and this gem, but we cannot release a version with support for that until they do first. I wouldn't recommend in production, but you can use this gem with DTA both in master and it should work with rails 7. See possible issues with doing that in https://github.com/lynndylanhurley/devise_token_auth/pull/1517#issuecomment-1003751213

**Comment by openscript** [comment]
@mcelicalderon Thank you very much for explaining :)

**Comment by mcelicalderon** [comment]
@maszuDEV yes, we removed support from master in order to make the release process easier. We don't have a good estimate on when DTA will release a version that supports Rails 7. When they do, it will be very simple here to add back the support. In the mean time if you were already using master to get Rails 7 support, I would recommend looking at this commit https://github.com/graphql-devise/graphql_devise/commit/f6435cf8486c8854d6c9034c3637a1ff37c6c194 if you need to add it back, perhaps we can create a separate branch with that if it will help you

**Comment by mcelicalderon** [comment]
@maszuDEV, created https://github.com/graphql-devise/graphql_devise/tree/rails-7-support to keep support for Rails 7

**Comment by maszuDEV** [comment]
> @maszuDEV, created https://github.com/graphql-devise/graphql_devise/tree/rails-7-support to keep support for Rails 7 in a branch

thank you very much ❤️

**Comment by CodySwannGT** [comment]
DTA v1.2.1 was just released with Rails 7!

**Comment by mcelicalderon** [comment]
Whaaat? That's awesome. Thank you for the ping, @CodySwannGT! I'm on it

**Comment by mcelicalderon** [comment]
Closing this one as [v1.1.0](https://rubygems.org/gems/graphql_devise/versions/1.1.0) was released. Thank you all for your patience!

**Comment by openscript** [comment]
@mcelicalderon Thank you for acting so quickly and all your work on this project!!!

**Comment by mcelicalderon** [comment]
Since master and the released version already support rails 7, please stop using the branch https://github.com/graphql-devise/graphql_devise/tree/rails-7-support so I can remove it in the near future

**Comment by SamuelAierizer** [comment]
Do you plan to support 7.2 as well? I think DTA finally has an official release that supports it: https://github.com/lynndylanhurley/devise_token_auth/issues/1637

**Comment by mcelicalderon** [comment]
Support for Rails 7.2 was merged into master in https://github.com/graphql-devise/graphql_devise/pull/278
